### ISSUE TITLE
Collision System

### ISSUE THREAD

**Issue by matteobruni** [matteobruni/tsparticles#104]
The particles should have a collision system, controlled by an option.

The collision can have various modes:
- **Merge**: The particles colliding should create a larger one
- **Bounce**: The particles colliding will bounce away
- **Destroy**: The particles colliding will destroy

**Merge Mode**: The larger one should be the one remaining and the one that become larger

**Destroy Mode**: The smaller particles could destroy against a larger one which remain intact or could become smaller. Two same size particles should destroy themself

**Comment by matteobruni** [comment]
The minimal collision system was already there, it's the bounce option in the particles.move section. An existing bug (of the original project) didn't let the particles bounce between themself if they haven't the attract o lineLinked property enabled. In the 1.8.1 release will be fixed this behavior. Merge and Destroy will be studied after.

**Comment by matteobruni** [comment]
If a particle will be blocked by all sides its velocity will become 0.
If a particle will be bounced and its velocity is 0 it’ll become the other particle velocity.
The bounce overlap will test every side, if none the particle will be blocked and will be in the status described some lines above

**Comment by matteobruni** [comment]
The base bounce is now ready for 1.13.0, merge and destroy will be moved to another issue for a future release
